### PR TITLE
port live-reload server functionality from phoenix

### PR DESCRIPTION
**Pull request by eastridge**

### CODE REVIEW

File: lib/live-reload.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+var io = require('socket.io'),
+  watch = require('watch'),
```
Comment (kpdecker) [inline lib/live-reload.js]: Line up vars

File: lib/live-reload.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  path = require('path'),
+  config = require('./config'),
+  debounce = function(func, wait) {
```
Comment (kpdecker) [inline lib/live-reload.js]: Use `function debounce()` format.

File: lib/live-reload.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      timeout = setTimeout(later, wait);
+    };
+  };
+
+module.exports = function(app, secureApp) {
+  var listener = io.listen(app);
+  listener.set('log level', 1);
+  
+  var secureListener = io.listen(secureApp);
+  secureListener.set('log level', 1);
```
Comment (kpdecker) [inline lib/live-reload.js]: `secureApp` will be null in heroku mode. Need to account for this. I'm also unsure if the apache/nginx proxy that is in front of the instance under heroku. My impl knowledge of websockets is low :(

Comment (eastridge) [inline lib/live-reload.js]: I've added a conditional around this in latest, but I don't think you'd ever run it with a `--live-reload` argument in heroku anyway, right?

**Pull request merged**